### PR TITLE
update for generalized constraint warnings

### LINKED ISSUE #492: [CT-2216] [Snowflake] Remove constraints_check warning in get_columns_spec_ddl
Core issue: https://github.com/dbt-labs/dbt-core/issues/7067

`dbt-snowflake` can use the [default implementation of `constraint_support](https://github.com/dbt-labs/dbt-core/issues/7067#issuecomment-1447281379) to model its enforced/support status of constraints. 

Once the warning mechanism has been generalized by https://github.com/dbt-labs/dbt-core/issues/7067, we can remove the warning raised in `get_columns_spec_ddl`: https://github.com/dbt-labs/dbt-snowflake/pull/341/files#diff-c96dce207e585a460348ff3f7fbd4c39d5822de2b13e2443648c0d8948066178R13-R15

### PR DESCRIPTION
**Pull request by emmyoop**
resolves #492

### Description

Uses centralized warnings from dbt-core for constraints.

### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-snowflake/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [ ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-snowflake/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

File: dbt/adapters/snowflake/impl.py [hunk 1]
```diff
@@ -246,17 +254,3 @@ def submit_python_job(self, parsed_model: dict, compiled_code: str):
 
     def valid_incremental_strategies(self):
         return ["append", "merge", "delete+insert"]
-
-    @classmethod
-    def render_column_constraint(cls, constraint: ColumnLevelConstraint) -> str:
-        if constraint.type == ConstraintType.check:
```
Comment (mikealfare) [inline dbt/adapters/snowflake/impl.py:252]: Maybe this is over-engineering it, but can we use `CONSTRAINT_SUPPORT`? So this:
```python
if self.CONSTRAINT_SUPPORT[constraint.type] == ConstraintSupport.NON_SUPPORTED:
    return ""
return super().render_column_constraint(constraint)
```
For that matter, can we just put this in `dbt-core`? Then we wouldn't need to override this method at all.

File: dbt/adapters/snowflake/impl.py [hunk 2]
```diff
@@ -246,17 +254,3 @@ def submit_python_job(self, parsed_model: dict, compiled_code: str):
...
-            return ""  # check not supported by snowflake
-        else:
-            return super().render_column_constraint(constraint)
-
-    @classmethod
-    def render_model_constraint(cls, constraint: ModelLevelConstraint) -> Optional[str]:
```
Comment (mikealfare) [inline dbt/adapters/snowflake/impl.py:258]: Same as above.

File: dbt/adapters/snowflake/impl.py (hunk 1 above)
Comment (VersusFacit) [reply dbt/adapters/snowflake/impl.py:252]: I'm not exactly sure what you mean by override – it seems that they're removing these methods entirely right?

**Review by peterallenwebb — APPROVED** [review]

File: dbt/adapters/snowflake/impl.py (hunk 1 above)
Comment (mikealfare) [reply dbt/adapters/snowflake/impl.py:252]: I don't read so gud

Comment (mikealfare) [reply dbt/adapters/snowflake/impl.py:252]: Apparently if you stare at code long enough, you forget what red/green and left/right mean.

**Review by mikealfare — APPROVED** [review]

**Pull request merged**